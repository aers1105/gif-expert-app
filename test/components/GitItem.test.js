const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe('Prueba de <GetItem />', () => {

    const
        title = 'Hola',
        url = 'http://localhost/EsteEsElUrl';
        
    test('Debe hacer match con el snapshot.', () => {


        const { container } = render( <GifItem title={title} url={url} /> );

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado.', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe mostrar el titulo del componente.', () => {

        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();

    })
});