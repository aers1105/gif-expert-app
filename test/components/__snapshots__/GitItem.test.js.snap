// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GetItem /> Debe hacer match con el snapshot. 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="http://localhost/EsteEsElUrl"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
